kimchi/src/folding/instance_witness.rs

```diff
@@ -0,0 +1,157 @@
+use super::{CommitmentCurve, One, PolyComm};
+
+pub trait InstanceTrait<G: CommitmentCurve>: Sized {
+    ///should return a linear combination
+    fn combine(a: Self, b: Self, challenge: G::ScalarField) -> Self;
+    fn relax(self, zero_commit: PolyComm<G>) -> RelaxedInstance<G, Self> {
+        let instance = self;
+        RelaxedInstance {
+            instance,
+            u: G::ScalarField::one(),
+            error_commitment: zero_commit,
+        }
+    }
+}
+pub trait WitnessTrait<G: CommitmentCurve>: Sized {
+    fn witness(&self, i: usize) -> &Vec<G::ScalarField>;
+    ///number of rows, or size of domain
+    fn rows(&self) -> usize;
+    fn witness_ext(&self, i: usize) -> &Vec<G::ScalarField>;
+    ///should return a linear combination
+    fn combine(a: Self, b: Self, challenge: G::ScalarField) -> Self;
+
+    fn relax(self, zero_poly: &[G::ScalarField]) -> RelaxedWitness<G, Self> {
+        let witness = self;
+        RelaxedWitness {
+            witness,
+            error_vec: zero_poly.to_vec(),
+        }
+    }
+}
+pub struct RelaxedInstance<G: CommitmentCurve, I: InstanceTrait<G>> {
+    instance: I,
+    pub u: G::ScalarField,
+    error_commitment: PolyComm<G>,
+}
+pub struct RelaxedWitness<G: CommitmentCurve, W: WitnessTrait<G>> {
```

Comment:
What is a relaxed witness and instance?

Reply:
Aren't they just the ones with the error terms allowed?